cmd/epp/main.go

```diff
@@ -247,62 +243,6 @@ func registerHealthServer(mgr manager.Manager, logger logr.Logger, ds datastore.
 	return nil
 }
 
-// registerMetricsHandler adds the metrics HTTP handler as a Runnable to the given manager.
-func registerMetricsHandler(mgr manager.Manager, port int, cfg *rest.Config, ds datastore.Datastore) error {
-	metrics.Register()
-	legacyregistry.CustomMustRegister(collectors.NewInferencePoolMetricsCollector(ds))
```

Comment:
How will the EPP expose InferencePool metrics with `NewInferencePoolMetricsCollector` removed?

Reply:
+1 that we need to keep the collector because the per pod queue size metric uses pod name as metric labels which will blow up the cardinality if it is not exported from a collector: https://github.com/kubernetes-sigs/gateway-api-inference-extension/blob/main/pkg/epp/metrics/collectors/inference_pool.go#L26

Reply:
Thanks for catching this. I need to figure out how to refactor it using `sigs.k8s.io/controller-runtime/pkg/metrics`

Reply:
Hi @JeffLuoo , pls review again for your convenience. Thanks.